pkg/controller/utils/clientwrapper.go

```diff
@@ -117,6 +117,15 @@ func (cmt *ControllerMetricsTripper) RoundTrip(req *http.Request) (*http.Respons
 	if err == nil && pathErr == nil {
 		metricKubeClientRequests.WithLabelValues(cmt.Controller.String(), req.Method, path, remoteStr, resp.Status).Inc()
 		metricKubeClientRequestSeconds.WithLabelValues(cmt.Controller.String(), req.Method, path, remoteStr, resp.Status).Observe(applyTime)
+		if applyTime >= 5.0 {
+			log.WithFields(log.Fields{
+				"controller": cmt.Controller.String(),
+				"method":     req.Method,
+				"path":       path,
+				"remote":     remoteStr,
+				"status":     resp.Status,
+			}).Warn("slow client request")
```

Comment:
is the log message really more useful than tracking this in a separate timeseries(metrics)?

Reply:
Technically it is represented in the timeseries we have now, with Suhani's help today I found 

```
sum without(job, endpoint, pod, service, instance)(hive_kube_client_request_seconds_bucket{remote="true",le="+Inf"} - ignoring(le) hive_kube_client_request_seconds_bucket{remote="true",le="5"} > 0)
```

Which gives us the counter for between 5s and infinity, which we could easily use with rate as well. However I would still like this in place for Thursday's test because it can help if we get into debugging the logs of a slow clustersync reconcile loop, sadly this logger won't have the reconcileID but it still should be possible to pinpoint via the controller and may help us determine if we're slow locally or remotely.